src/Simplex/Messaging/Server/StoreLog.hs

```diff
@@ -62,29 +62,31 @@ instance StrEncoding QueueRec where
       ]
       <> maybe "" notifierStr notifier
     where
-      notifierStr (nId, nKey) = " nid=" <> strEncode nId <> " nk=" <> strEncode nKey
+      notifierStr NtfCreds {notifierId, notifierKey, rcvNtfDhSecret} = " nid=" <> strEncode notifierId <> " nk=" <> strEncode notifierKey <> " ndh=" <> strEncode rcvNtfDhSecret
 
   strP = do
     recipientId <- "rid=" *> strP_
     recipientKey <- "rk=" *> strP_
     rcvDhSecret <- "rdh=" *> strP_
     senderId <- "sid=" *> strP_
     senderKey <- "sk=" *> strP
-    notifier <- optional $ (,) <$> (" nid=" *> strP_) <*> ("nk=" *> strP)
+    notifier <- optional $ NtfCreds <$> (" nid=" *> strP_) <*> ("nk=" *> strP_) <*> ("ndh=" *> strP)
```

Comment:
this one can also use NtfCreds instance, it would be just `notifier=` with space-separated values